docs/source/daml/intro/11_Testing.rst

```diff
@@ -0,0 +1,283 @@
+
+Script Service CLI Testing
+..........................
+
+In your chapter 8 project, run ``daml test``. This runs the same Scripts that are run in the IDE on the same in-memory Ledger called the Script Service. Instead of views of the resulting ledger, it outputs successes and failures.
+
+.. code-block:: none
+
+  daml/Test/Intro/Asset/TradeSetup.daml:setupRoles: ok, 2 active contracts, 4 transactions.
+  daml/Test/Intro/Asset/TradeSetup.daml:test_issuance: ok, 3 active contracts, 5 transactions.
+  daml/Test/Intro/Asset/TradeSetup.daml:tradeSetup: ok, 6 active contracts, 10 transactions.
+  daml/Test/Intro/Asset/MultiTrade.daml:testMultiTrade: ok, 12 active contracts, 28 transactions.
+
+If you add the command line flag ``--junit``, ``daml test`` can write the results in jUnit format, which is useful for automated testing.
+
+Sandbox CLI Testing
+...................
+
+The next step up in terms of how "real" the tests are is the command ``daml test-script``. Unlike ``daml test``, these tests are performed via a Ledger API against a real Ledger. If no ledger host and port are provided, a temporary Sandbox is started to run the tests against.
+
+In the same project, make sure you have compiled the contracts (``daml build -o multi-trade.dar``), then run ``daml test-script --dar multi-trade.dar``. You'll see a lot more output this time, starting with a Sandbox being started up, followed by DAML Scrpipt outputs. Each script should finish with a message of the type ``Test.Intro.Asset.MultiTrade.daml:testMultiTrade SUCCESS``.
+
+Script Against a Running Ledger
+...............................
+
+The last thing we can do with DAML Script is to run against a live ledger through the Ledger API. To see this in action, we'll initialize a Sandbox Ledger with the trade initialization data from chapter 7 model. Go to your chapter 8 project.
+
+#. Start a sandbox with ``daml sandbox``.
+#. Upload the DAR file form the chapter 8 project by running ``daml ledger upload-dar multi-trade.dar``.
+#. Run the v1Setup Script with ``daml script --dar multi-trade.dar --ledger-host localhost --ledger-port 6865 --script-name Test.Intro.Asset.TradeSetup:tradeSetup``
+
+By default, this doesn't log much so if it returns with code 0, you can assume it was successful. But how can we check?
+
+DAML Navigator
+~~~~~~~~~~~~~~
+
+:doc:`/tools/navigator/index` is a UI that runs against a Ledger API and allows interaction with contracts. The easiest way to start it against an already running ledger is using ``daml ledger navigator``. If successful, it will tell you ``Frontend running at http://localhost:4000.``. Navigate to that URL and log in as Alice. You'll see the three contracts created by the setup script: One ``AssetHolder`` contract for each bank, and one ``Asset``. If so, the script was successful.
+
+We now want establish more relationships and issue assets. We could do that using Navigator, but it would be some work.
+
+DAML Script with Inputs
```

Comment:
I do agree that this is a crucial feature. What I disagree with is that it needs to be mentioned here. The DAML Script docs are very approachable (and if they aren’t I’ll gladly accept PRs or suggestions to improve that). Replicating here just so you have only a single place that people need to read through for certification is making the overall structure of our docs worse.